### DISCUSSION TITLE
Why is there no `Add` method available for Dictionary that takes a `KeyValuePair` as its parameter?

### DISCUSSION THREAD

**Discussion by egvijayanand** [dotnet/sdk#45425] [General]
Why is there no `Add` method available for Dictionary that takes a `KeyValuePair` as its parameter?

Any specific reason?

```cs
public void Add(KeyValuePair<TKey, TValue> kvp) => Add(kvp.Key, kvp.Value);
```

**Comment by egvijayanand** [comment]
I'm trying to define a Dictionary in XAML, but without a `ContentProperty` defined for a Dictionary, I cannot specify its contents.

So, I'm trying to use KeyValuePair for this. However, since no Add method accepts a KeyValuePair as its argument, I am unable to proceed.

**Comment by KalleOlaviNiemitalo** [comment]
Dictionary\<TKey, TValue\> has an Add(KeyValuePair\<TKey, TValue\>) method as an explicit interface implementation for ICollection\<KeyValuePair\<TKey, TValue\>\>: <https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.system-collections-generic-icollection-system-collections-generic-keyvaluepair-tkey-tvalue---add?view=net-9.0#system-collections-generic-dictionary-2-system-collections-generic-icollection((system-collections-generic-keyvaluepair((tkey@tvalue))))-add(system-collections-generic-keyvaluepair((-0-1)))>

But I don't know whether that can be used in XAML.

**Comment by egvijayanand** [comment]
Yes, it's implemented explicitly. It's quite unfortunate.

**Comment by KalleOlaviNiemitalo** [comment]
Can you use [ResourceDictionary](https://learn.microsoft.com/dotnet/api/system.windows.resourcedictionary?view=windowsdesktop-9.0) instead?  It is not generic though, and it is defined in PresentationFramework.dll rather than System.Xaml.dll, so might not be appropriate if you are using XAML for non-UI purposes.
